iiasa/message-ix-models
# Transport improvements from 2024-W49

- Items for SSP 2024/ScenarioMIP.
  - [x] Update base scenario URLs from e.g. `ixmp://ixmp-dev/SSP_dev_SSP1_v0.1_Blv0.18/baseline_prep_lu_bkp_solved_materials` to e.g. `ixmp://ixmp-dev/SSP_SSP1_v1.1/baseline_DEFAULT_step_13`.
  - [x] Constrain t=conm_ar (a P AIR technology) to a low/zero share in 2020. This technology was (a) taking a large share in y=2020 without constraints and (b) because not present at all in IEA EWEB for some regions, was being scaled with 1.0, so passing through as a large value.
  - ~[ ]~ Next PR: Use a smaller `growth_activity_up` value for (t=ICAm_ptrp, n=R12_RCPA). The technology constraint as of 2020 is effective, but this tech/region grows too rapidly to 2025 and beyond. (per discussion with @r-aneeque 2024-12-04.)
  - ~[ ]~ Next PR: Adjust efficiencies of LDV PHEV technologies using duty cycle data.
  - ~[ ]~ Next PR: Ensure `growth_activity_up` and `act-non_ldv.csv` values are set for `F ROAD` technologies with, at least, c=gas inputs. This is to avoid the behaviour (observed by @r-aneeque) that the model quickly switches into these in 2025–2030. Rework `.transport.Config.constraint` to simplify the setting of these constraints.
- Items for aviation emissions postprocessing:
  - [x] Implement method 'B'.
  - [x] #230

This branch ~is currently~ was temporarily triggering the message_data transport runs using iiasa/message_data#582, which limits to a subset of all scenarios.

## How to review

Per aviation emissions:
- View the diff (click 'Files changed' above), particularly the file `message_ix_models/data/transport/emi-intensity.csv`.
- View the [**preview build**](https://iiasa-energy-program-message-ix--259.com.readthedocs.build/projects/models/en/259/) of the documentation, in particular the page “Shared Socioeconomic Pathways (project.ssp)” and section “2024 update > transport”, [here](https://iiasa-energy-program-message-ix--259.com.readthedocs.build/projects/models/en/259/project/ssp.html#update).
- Confirm the code runs without error on the files in [ECE SharePoint > Documents > SharedSocioeconomicPathways2023 > Scenario_Vetting](https://iiasahub.sharepoint.com/sites/eceprog/Shared%20Documents/Forms/AllItems.aspx?csf=1&web=1&e=APKv0Z&CID=23fa0a51%2Dc303%2D4381%2D8c6d%2D143305cbc5a1&FolderCTID=0x012000AA9481BF7BE9264E85B14105F7F082FF&id=%2Fsites%2Feceprog%2FShared%20Documents%2FSharedSocioEconomicPathways2023%2FScenario%5FVetting&viewid=956acd8a%2De1e7%2D4ae9%2Dab1b%2D0506911bae11) (for instance, the subdirectory v2.1_Internal_version_Dec13_2024 > Reporting_output.
- View the output and note any qualitative or quantitative issues to be resolved.

**More TBA**



## PR checklist
